app/file.go

```diff
@@ -1070,113 +1039,75 @@ func (a *App) HandleImages(previewPathList []string, thumbnailPathList []string,
 	wg := new(sync.WaitGroup)
 
 	for i := range fileData {
-		img, _, _ := prepareImage(fileData[i])
-		if img != nil {
-			wg.Add(2)
-			go func(img image.Image, path string) {
-				defer wg.Done()
-				a.generateThumbnailImage(img, path)
-			}(img, thumbnailPathList[i])
-
-			go func(img image.Image, path string) {
-				defer wg.Done()
-				a.generatePreviewImage(img, path)
-			}(img, previewPathList[i])
+		img, release, err := prepareImage(a.srv.imgDecoder, bytes.NewReader(fileData[i]))
+		if err != nil {
+			mlog.Debug("Failed to prepare image", mlog.Err(err))
+			continue
 		}
+		wg.Add(2)
+		go func(img image.Image, path string) {
+			defer wg.Done()
+			a.generateThumbnailImage(img, path)
+		}(img, thumbnailPathList[i])
+
+		go func(img image.Image, path string) {
+			defer wg.Done()
+			a.generatePreviewImage(img, path)
+		}(img, previewPathList[i])
+
+		wg.Wait()
+		release()
 	}
-	wg.Wait()
 }
 
-func prepareImage(fileData []byte) (image.Image, int, int) {
+func prepareImage(imgDecoder *imaging.Decoder, imgData io.ReadSeeker) (img image.Image, release func(), err error) {
```

Comment:
I have a question here, would it make sense to convert this function to a `imaging.Decoder` method? Seems like we are doing an imaging operation here and would it make sense to give this responsibility to `imaging` package?

Reply:
I've thought about that but eventually felt this was a bit too app specific to fit in the package which at some point might even go under `shared` as a standalone service/library.

Reply:
I see. Thanks for the explanation 👍